lib/Compilation.js

```diff
@@ -2294,7 +2294,7 @@ class Compilation extends Tapable {
 			const module = modules[i];
 			const moduleHash = createHash(hashFunction);
 			module.updateHash(moduleHash);
-			module.hash = moduleHash.digest(hashDigest);
+			module.hash = /** @type {string} */ (moduleHash.digest(hashDigest));
```

Comment:
casting to `string` since `digest` method can return `Buffer` if `encoding` does not provided